lnwallet/channel.go

```diff
@@ -1852,6 +1912,16 @@ func (lc *LightningChannel) ReceiveNewCommitment(rawSig []byte) error {
 	localCommitmentView.sig = rawSig
 	lc.localCommitChain.addCommitment(localCommitmentView)
 
+	// If we are not channel initiator, then the commitment just received would've
+	// signed any received fee update since last commitment. Mark any such fee
+	// update as pending ACK (so we remember to ACK it on our next commitment),
+	// and set pendingFeeUpdate to nil. We can do this since we won't receive any
+	// new commitment before ACKing.
+	if !lc.channelState.IsInitiator {
+		lc.pendingAckFeeUpdate = lc.pendingFeeUpdate
```

Comment:
The commitment with the updated fee is being added to the commit chain, and then persisted to disk when `UpdateCommitment` is called.